plugins/checksum/checksum.c

```diff
@@ -31,49 +31,46 @@ static char name[] = "Checksum";
 static char desc[] = "Calculate checksum for DCC file transfers";
 static char version[] = "4.0";
 
-static void
-print_sha256_result (hexchat_context *ctx, gboolean send_message, const char *checksum, const char *filename, GError *error)
-{
-	/* Context may have been destroyed. */
-	/* FIXME: This could still send the PRIVMSG even if the context was closed. */
-	if (!hexchat_set_context (ph, ctx))
-		return;
 
-	if (error)
-		hexchat_printf (ph, "Failed to create checksum for %s: %s", filename, error->message);
-	else if (send_message)
-		hexchat_commandf (ph, "quote PRIVMSG %s :SHA-256 checksum for %s (remote): %s", hexchat_get_info (ph, "channel"), filename, checksum);
-	else
-		hexchat_printf (ph, "SHA-256 checksum for %s (local): %s\n", filename, checksum);
-}
+typedef struct {
+	hexchat_context *ctx;
+	gboolean send_message;
+} ChecksumCallbackInfo;
+
 
-/* TODO: We could put more info in task data and share the same callback. */
 static void
-on_received_file_sha256_complete (GFile *file, GAsyncResult *result, gpointer user_data)
+print_sha256_result (ChecksumCallbackInfo *info, const char *checksum, const char *filename, GError *error)
 {
-	hexchat_context *ctx = user_data;
-	GError *error = NULL;
-	char *sha256 = NULL;
-	const char *filename = g_task_get_task_data (G_TASK (result));
 
-	sha256 = g_task_propagate_pointer (G_TASK (result), &error);
-	print_sha256_result (ctx, FALSE, sha256, filename, error);
+	/* Context may have never been set or destroyed.*/
+	if(!info->ctx  || !hexchat_set_context (ph, info->ctx)){
+		// So then we get the next best available channel, since we always want to print at least somewhere, it's fine
+		hexchat_context * ctx = hexchat_find_context (ph, NULL, NULL);
+		hexchat_set_context (ph, ctx);
+	}
 
-	g_free (sha256);
-	g_clear_error (&error);
+	if (error){
+		hexchat_printf (ph, "Failed to create checksum for %s: %s\n", filename, error->message);
+	}
+	else if (info->send_message){
+		hexchat_commandf (ph, "quote PRIVMSG %s :SHA-256 checksum for %s (remote): %s", hexchat_get_info (ph, "channel"), filename, checksum);
```

Comment:
I think your `hexchat_find_context()` above creates the side-effect that it *may* change to a different network's context. This would then send a message to a user on a different network.